### PR TITLE
PedigreeViz - hide overflowing detail text unless hovered or selected

### PR DESCRIPTION
**Pull request by alexkb0009**
[Video Demo](https://gyazo.com/631a5f14d3047d9a4805b9e35cffcc7b)

### CODE REVIEW

File: src/encoded/static/components/viz/PedigreeViz/IndividualsLayer.js [hunk 1]
```diff
@@ -139,34 +148,154 @@ export class IndividualDiv extends React.PureComponent {
...
+        // Additional margin-top for .detail-text may be supplied in (S)CSS sheets.
+        transform: `translate3d(-${ individualXSpacing / 2 }px,0,0)`,
+        bottom: individualHeight * 1.25,
+        fontSize: Math.min(textScale, 2) + "rem",
+        transformOrigin: `${ individualWidth / 2 }px 100%`
+    };
+
+    return (
+        <div className="above-node-text" data-describing="ancestry" style={textContainerStyle}>
+            { showAncestry.join(" • ") }
```
Comment (alexkb0009) [inline src/encoded/static/components/viz/PedigreeViz/IndividualsLayer.js:282]: Wonder if better separator or formatting of some kind could be better here.. given that sometimes (depending on length+zoom level) it wraps to next to line.

File: src/encoded/static/components/viz/PedigreeViz/IndividualsLayer.js [hunk 2]
```diff
@@ -139,34 +148,154 @@ export class IndividualDiv extends React.PureComponent {
...
+    if (heightIndex !== maxHeightIndex) return null;
+    if (ancestry.length === 0) return null;
+
+    // TODO return null if all partners in relationship have same ancestry contents
+    // and then render out above the relationship node.
+
+    const moreAncestryPresent = ancestry.length > 3;
+    const showAncestry = moreAncestryPresent ? ancestry.slice(0, 3) : ancestry; // Up to 3 shown.
+
+    const textContainerStyle = {
```
Comment (alexkb0009) [inline src/encoded/static/components/viz/PedigreeViz/IndividualsLayer.js:270]: In future could maybe move some more of this into plain CSS..
This style object is essentially same for each such node so theoretically could be memoized at a higher level once and then passed down to all IndividualDiv > AboveNodeText.. 🤔

Comment (alexkb0009) [reply src/encoded/static/components/viz/PedigreeViz/IndividualsLayer.js]: ..memoization part done

**Review by Bianca-Morris — CHANGES_REQUESTED** [review]
Looks like ancestry isn't showing up above the individual nodes anymore? And in the side-bar, the ancestry text is pretty jumbled together. https://gyazo.com/53f01add0c5e289dc045041bd58c0ac6

Phenotype overflow looks great, though.

**Comment by alexkb0009** [comment]
> Looks like ancestry isn't showing up above the individual nodes anymore? And in the side-bar, the ancestry text is pretty jumbled together. https://gyazo.com/53f01add0c5e289dc045041bd58c0ac6
> 
> Phenotype overflow looks great, though.

[Fixed the side-bar text](https://gyazo.com/7778b280856621cad85e19379943de35); for ancestry above nodes, it's only displayed for the terminal ancestors and not any child nodes (would overlap and/or prevent showing as much below-node detail text for nodes).

**Pull request merged**